modules/ding_adgangsplatformen/ding_adgangsplatformen.module

```diff
@@ -334,6 +334,18 @@ function ding_adgangsplatformen_callback() {
       $user_info = $response->getBody()->getContents();
       $user_info = drupal_json_decode($user_info);
 
+      //
+      // HACK: this invalidates the access token and should be removed later on.
+      //       This only exists to get around pre-authentication issues
+      //       currently in FBS.
+      //
+      $request = $provider->getAuthenticatedRequest('DELETE', 'https://login.bib.dk/oauth/revoke', $access_token);
+      $response = $provider->getResponse($request);
+      if ($response->getStatusCode() != 200) {
+        watchdog('ding_adgangsplatformen', "https://login.bib.dk/oauth/revoke returned status code: %code", array('%code' => $response->getStatusCode()), WATCHDOG_CRITICAL);
+      }
+      unset($_SESSION['oauth2token']);
```

Comment:
@cableman I was looking at this and wondering why you unset the oauth2token here?

Reply:
Because this is a hack and the token is valid for a very long time. So if the token is not revoked and removed and it's leaked. The end users library information include username and password can be fetch with a simple curl call.

So so long that adgangsplatformen contains the user pincode in userinfo... this is a hack.

Reply:
So this also involves a temporarily hack in adgangsplatformen that will return pincode?

Reply:
Yes.

This breaks the whole security model behind oAuth2. So it should be removed as soon as possible. But as FBS changes don't happen over night this is the compromise decided by  DDB. The whole thing links back to the issue that FBS can't login all users types via pre-authenticate.

Reply:
Ok I see :) thanks for the explanation @cableman 👍